Tempo/Views/ExpandedPlayerView.swift

```diff
@@ -0,0 +1,314 @@
+		post?.player.progress = progress
+		
+		progressView.setNeedsDisplay()
+	}
+	
+	func collapsePan(gesture: UIPanGestureRecognizer) {
+		let translation = gesture.translationInView(self)
+		if gesture.state == .Began || gesture.state == .Changed {
+			let maxCenter = UIScreen.mainScreen().bounds.height - height/2
+			
+			if translation.y > 0 || self.center.y > maxCenter {
+				if self.center.y + translation.y < maxCenter {
+					self.center.y = maxCenter
+				} else {
+					self.center.y = self.center.y + translation.y
+				}
+			}
+			gesture.setTranslation(CGPointMake(0,0), inView: self)
+		}
+		
+		if gesture.state == .Ended {
+			let velocity = gesture.velocityInView(self)
+			parentNav?.animateExpandedCell(velocity.y < 0)
+			initialPanView = nil
+		}
+		setNeedsDisplay()
+	}
+	
+	func updatePlayToggleButton() {
+		if let selectedPost = post {
+			let name = selectedPost.player.isPlaying ? "pause-red" : "play-red"
+			progressView.setUpTimer()
+			playToggleButton.setBackgroundImage(UIImage(named: name), forState: .Normal)
+		}
+	}
+	
+	private func updateAddButton() {
+		addButton.userInteractionEnabled = false
+		addButton.hidden = true
+		if let _ = post {
+			SpotifyController.sharedController.spotifyIsAvailable { success in
+				if success {
+					self.addButton.hidden = false
+					self.addButton.userInteractionEnabled = true
+					self.addButtonImage.image = (self.songStatus == .Saved) ? UIImage(named: "check") : UIImage(named: "plus")
+				}
+			}
+		}
+		
+	}
+	
+	func updateLikeButton() {
+		if let selectedPost = post {
+			if postsLikable {
+				likeButton.userInteractionEnabled = true
+				let name = selectedPost.isLiked ? "filled-heart" : "empty-heart"
+				likeButtonImage.image = UIImage(named: name)
+			} else {
+				likeButton.userInteractionEnabled = false
+				likeButtonImage.image = UIImage(named: "empty-heart")
+			}
+		}
+	}
+	
+//	@IBAction func playToggleButtonClicked(sender: UIButton) {
+//		if let selectedPost = post {
+//			selectedPost.player.togglePlaying()
+//			updatePlayToggleButton()
+//		}
+//	}
+	
+//	@IBAction func addButtonClicked(sender: UIButton) {
+//		if songStatus == .NotSaved {
+//			SpotifyController.sharedController.saveSpotifyTrack(post!) { success in
+//				if success {
+//					self.addButtonImage.image = UIImage(named: "check")
+//					self.songStatus = .Saved
+//				}
+//			}
+//		} else if songStatus == .Saved {
+//			SpotifyController.sharedController.removeSavedSpotifyTrack(post!) { success in
+//				if success {
+//					self.addButtonImage.image = UIImage(named: "plus")
+//					self.songStatus = .NotSaved
+//				}
+//			}
+//		}
+//	}
+	
+	
+//	@IBAction func likeButtonClicked(sender: UIButton) {
+//		if let selectedPost = post {
+//			selectedPost.toggleLike()
+//			updateLikeButton()
+//			NSNotificationCenter.defaultCenter().postNotificationName(PostLikedStatusChangeNotification, object: self)
+//		}
+//	}
```

Comment:
Do we still need references to lines 270 - 302? If not, let's delete the commented code